src/MDMTransaction.h

```diff
@@ -27,9 +27,36 @@ NS_SWIFT_NAME(Transaction)
 
 #pragma mark Adding plans to a transaction
 
-/** Associate an plan with a given target. */
+/**
+ Associates a plan with a given target.
+
+ @param plan The plan to add to this transaction.
+ @param target The target on which the plan can operate.
+*/
+- (void)addPlan:(nonnull id<MDMPlan>)plan
+       toTarget:(nonnull id)target
+  NS_SWIFT_NAME(add(plan:to:));
+
+/**
+ Associates a named plan with a given target.
+ 
+ @param plan The plan to add to this transaction.
+ @param target The target on which the plan can operate.
+ @param name String identifier for the plan.
+ */
 - (void)addPlan:(nonnull id<MDMPlan>)plan
        toTarget:(nonnull id)target
-    NS_SWIFT_NAME(add(plan:to:));
+          named:(nonnull NSString *)name
+  NS_SWIFT_NAME(add(plan:to:withName:));
```

Comment:
Am appreciating that Swift API naming is encouraging thinking about these as sentences. In that manner "add this plan named 'interaction' to this target" is probably less ambiguous than "add this plan to this target named 'interaction'". In the latter case I can imagine confusion by assuming that the target is the thing being named.

Perhaps `addPlan:named:toTarget:` and `addPlan(_:named:to:)` will be cleaner overall? What do you think? I'll update the spec if we choose to go this route.